mars/tensor/expressions/indexing/getitem.py

```diff
@@ -34,29 +34,29 @@ class TensorIndex(Index, TensorOperandMixin):
     def __init__(self, dtype=None, sparse=False, **kw):
         super(TensorIndex, self).__init__(_dtype=dtype, _sparse=sparse, **kw)
 
-    @classmethod
-    def _handle_inputs(cls, inputs):
-        tensor, indexes = inputs
-        indexes_inputs = [ind for ind in indexes if isinstance(ind, (BaseWithKey, Entity))]
-        return [tensor] + indexes_inputs
+    def _handle_inputs(self, inputs):
+        if len(inputs) == 1:
```

Comment:
When `new_chunks` called by `GraphActor.get_executable_operand_dag`, the inputs can be more than 1, so the logic may crash when inputs' lengh > 1.

We can extend `new_chunks` and `new_tensors` by supporting add `indexes` param, and if `indexes` is provided, just set attribute `_indexes` with `indexes`, and finally do `_set_inputs`, if not, we will check if the op has the `_indexes` attributes.

Reply:
all done.